bundles/org.openhab.binding.tapocontrol/src/main/java/org/openhab/binding/tapocontrol/internal/TapoControlHandlerFactory.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.tapocontrol.internal;
+
+import static org.openhab.binding.tapocontrol.internal.constants.TapoBindingSettings.*;
+import static org.openhab.binding.tapocontrol.internal.constants.TapoThingConstants.*;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.openhab.binding.tapocontrol.internal.device.TapoBridgeHandler;
+import org.openhab.binding.tapocontrol.internal.device.TapoLightStrip;
+import org.openhab.binding.tapocontrol.internal.device.TapoSmartBulb;
+import org.openhab.binding.tapocontrol.internal.device.TapoSmartPlug;
+import org.openhab.binding.tapocontrol.internal.device.TapoUniversalDevice;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseThingHandlerFactory;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerFactory;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link TapoControlHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Christian Wild - Initial contribution
+ */
+@Component(service = ThingHandlerFactory.class, configurationPid = "binding.tapocontrol")
+@NonNullByDefault
+public class TapoControlHandlerFactory extends BaseThingHandlerFactory {
+    private final Logger logger = LoggerFactory.getLogger(TapoControlHandlerFactory.class);
+    private final Set<TapoBridgeHandler> accountHandlers = new HashSet<>();
+    private final HttpClient httpClient;
+
+    @Activate
+    public TapoControlHandlerFactory() {
+        // create new httpClient
+        httpClient = new HttpClient(new SslContextFactory.Client());
+        httpClient.setFollowRedirects(false);
+        httpClient.setMaxConnectionsPerDestination(HTTP_MAX_CONNECTIONS);
+        httpClient.setMaxRequestsQueuedPerDestination(HTTP_MAX_QUEUED_REQUESTS);
```

Comment:
What's the reason you need to disable follow redirects?

If you really need to disable thoses, you can create your own pre-configured HTTP client by https://www.openhab.org/javadoc/latest/org/openhab/core/io/net/http/httpclientfactory

Please stop your HTTP client in the deactivate method.

Reply:
Ok forgott to stop it. 
I got some communication problems using the common client from httpclientfactory.
So i used a new httpClient and those were solved. setFollowRedirects is disabled defaultly in the most examples and there's no need to use redirects

Reply:
I got some session (cookie) problems while device communications.